### ISSUE TITLE
MS: do not apply priority order to workflows without any input data

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9592]
**Impact of the bug**
ReqMgr2MS

**Describe the bug**
Only today I noticed that, in a bad scenario with many sites out of quota, we can hit the situation where most of the 500 workflows (set via `limitRequestsPerCycle`) will not go through in the ReqMgr2MS cycle, being retried in the next cycle. 

This will delay the other workflows sitting in `assigned` state, even if they have no input data to be placed. Reason for that is that we order workflows by priority, if high priority workflows don't manage to go to `staging`, then low prio will never be evaluated.

**How to reproduce it**
Steps can be extracted from the description above.

**Expected behavior**
Workflows without any kind of input data should not wait for the normal priority cycle to pick them up.
In other words, low priority workflows should have a chance to be evaluated as well. If they don't require any input dataset, there is no reason to keep holding them under the `assigned` status.

I see two options here:
a) every other cycle we go an extra mile and process beyond the `limitRequestsPerCycle` number of workflows, but only move forward with those without any input data.
b) every cycle, we consider `limitRequestsPerCycle` extra workflows not requiring any sort of input data.

It's not clear to me which approach will be the best (especially in terms of implementation) at this moment.

**Additional context and error message**
none
